### ISSUE TITLE
Set connection manually

### ISSUE THREAD

**Issue by rekrios** [cloudcreativity/laravel-json-api#373]
Hi. Maybe make available to change 'protected $connection;' without controller extending ?

**Comment by lindyhopchris** [comment]
Yes in principle that's fine... but can't do this on the `1.0` series because changing it to `public` will be breaking for all controllers that have extended the class and have it as `protected`.

Solution would be a public method `connection()` that allows you to set it. I would accept a PR for that if you wanted to do one.

**Comment by lindyhopchris** [comment]
When I wrote the above, I was assuming you would want to change this via some container binding: i.e. bind a class that did an override of that property.

However, maybe you meant you'd prefer to set it via config? If so, that might actually be a better solution. Let me know and might be quicker if I sort this out rather than you doing a PR.

**Comment by rekrios** [comment]
Maybe make both solutions ?
- init via config(if set, by default null)
- availability to set via connection()
?

**Comment by lindyhopchris** [comment]
Yeah, sounds like a plan! Think it also makes sense to allow toggling of the `useTransactions` property too in exactly the same way.

No need to send a PR... think this will be quicker for me to sort out.

**Comment by lindyhopchris** [comment]
@rekrios have made changes, can you give this a go?

Require the dev version of the package:

```bash
composer require cloudcreativity/laravel-json-api:1.x-dev
```

Then add this to your API's config file:

```php
    /*
    |--------------------------------------------------------------------------
    | Controllers
    |--------------------------------------------------------------------------
    |
    | The default JSON API controller wraps write operations in transactions.
    | You can customise the connection for the transaction here. Or if you
    | want to turn transactions off, set `transactions` to `false`.
    |
    */
    'controllers' => [
        'transactions' => true,
        'connection' => null,
    ],
```

And customise as required. Let me know how you get on as I won't tag it until I've had feedback!

**Comment by rekrios** [comment]
sr, didn't see, i will try soon

**Comment by lindyhopchris** [comment]
Released as `1.3`. Feel free to re-open if you find any problems.

**Comment by rekrios** [comment]
thx
